src/PrestaShopBundle/Form/Admin/Sell/Order/OrderMessageType.php

```diff
@@ -0,0 +1,115 @@
+ */
+
+namespace PrestaShopBundle\Form\Admin\Sell\Order;
+
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\CleanHtml;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+use PrestaShopBundle\Form\Admin\Type\TextWithLengthCounterType;
+use PrestaShopBundle\Translation\TranslatorAwareTrait;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * Helps to render messages block in orders view page.
+ */
+class OrderMessageType extends AbstractType
+{
+    use TranslatorAwareTrait;
+
+    /**
+     * @var FormChoiceProviderInterface
+     */
+    private $orderMessageNameChoiceProvider;
+
+    /**
+     * @var FormChoiceProviderInterface
+     */
+    private $orderMessageChoiceProvider;
+
+    public function __construct(
+        FormChoiceProviderInterface $orderMessageNameChoiceProvider,
+        FormChoiceProviderInterface $orderMessageChoiceProvider
+    ) {
+        $this->orderMessageNameChoiceProvider = $orderMessageNameChoiceProvider;
+        $this->orderMessageChoiceProvider = $orderMessageChoiceProvider;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('order_message', ChoiceType::class, [
+                'choices' => $this->orderMessageNameChoiceProvider->getChoices(),
+                'required' => false,
+            ])
+            ->add('is_displayed_to_customer', CheckboxType::class, [
+                'required' => false,
+                'label' => $this->trans('Display to customer?', [], 'Admin.Orderscustomers.Feature'),
+            ])
+            ->add('message', TextWithLengthCounterType::class, [
+                'input' => 'textarea',
+                'max_length' => 600,
+                'position' => 'after',
+                'constraints' => [
+                    new NotBlank([
+                        'message' => $this->trans(
+                            'This field cannot be empty', [], 'Admin.Notifications.Error'
+                        ),
+                    ]),
+                    new CleanHtml([
+                            'message' => $this->trans(
+                                'The %s field is not valid',
+                                [
+                                    sprintf('"%s"', $this->trans('Message', [], 'Admin.Global')),
+                                ],
+                                'Admin.Notifications.Error'
+                            ),
+                        ]
+                    ),
+                    new Length([
+                        'max' => 600,
```

Comment:
Looks like a nice candidate for a domain constant 🤔 or maybe it's only form-related ?